### PR TITLE
Integer: CLN: Move implementation of member functions to .cpp file.

### PR DESCRIPTION
**Pull request by aniemetz**
This moves the CLN implementation of member functions of class Integer from
the header to the .cpp file. This only moves code, and adds documentation for
previously undocumented or poorly documented functions.

Analogous to #5190.

### CODE REVIEW

File: src/util/integer_cln_imp.cpp [hunk 1]
```diff
@@ -53,41 +151,252 @@ Integer Integer::setBit(uint32_t i, bool value) const
...
+
+int Integer::sgn() const
+{
+  cln::cl_I sgn = cln::signum(d_value);
+  return cln::cl_I_to_int(sgn);
+}
+
+bool Integer::strictlyPositive() const { return sgn() > 0; }
+
+bool Integer::strictlyNegative() const { return sgn() < 0; }
```
Comment (nafur) [inline src/util/integer_cln_imp.cpp:362]:
```suggestion
bool Integer::strictlyNegative() const { return minusp(d_value); }
```

File: src/util/integer_cln_imp.cpp [hunk 2]
```diff
@@ -98,104 +407,159 @@ void Integer::readInt(const cln::cl_read_flags& flags,
...
     // including CLN v1.3.2.
-    // See http://www.ginac.de/CLN/cln.git/?a=commit;h=4a477b0cc3dd7fbfb23b25090ff8c8869c8fa21a for details.
+    // See
+    // http://www.ginac.de/CLN/cln.git/?a=commit;h=4a477b0cc3dd7fbfb23b25090ff8c8869c8fa21a
+    // for details.
     size_t pos = s.find_first_not_of('0');
-    if(pos == std::string::npos) {
+    if (pos == std::string::npos)
+    {
       d_value = read_integer(flags, "0", NULL, NULL);
```
Comment (nafur) [inline src/util/integer_cln_imp.cpp:420]:
```suggestion
      d_value = cln::read_integer(flags, "0", NULL, NULL);
```

File: src/util/integer_cln_imp.cpp [hunk 3]
```diff
@@ -98,104 +407,159 @@ void Integer::readInt(const cln::cl_read_flags& flags,
...
-    } else {
+    }
+    else
+    {
       const char* cstr = s.c_str();
       const char* start = cstr + pos;
       const char* end = cstr + s.length();
       d_value = read_integer(flags, start, end, NULL);
```
Comment (nafur) [inline src/util/integer_cln_imp.cpp:427]:
```suggestion
      d_value = cln::read_integer(flags, start, end, NULL);
```

**Review by nafur — APPROVED** [review]
Just a few minor improvements (using direct calls to `cln` instead of custom checks that may be a bit slower.

**Comment by aniemetz** [comment]
@nafur I feel like there is room for improvement in terms of performance both in the CLN and BMP Integer implementation. I'll add the changes. Additionally, eventually we also should have a closer look at the GMP implementation with that regard.

**Comment by aniemetz** [comment]
On second thought, let me add the changes in a new PR. This one only moves code.

**Pull request merged**